src/main/java/org/jitsi/jicofo/bridge/BridgeSelectionStrategy.java

```diff
@@ -67,79 +69,120 @@ public Bridge select(
+     * is in the least stressed group.
+     *
+     * Suppose for example that we have 6 bridges, each loaded from 1 to 6 that
+     * we want to split into 3 groups (imagine the groups names are "low-stressed",
+     * "medium-stressed" and "high-stressed" bridges):
+     *
+     * min = 1, max = 6, step = ceil(5 / 3)  = cel(1.66) = 2
+     * group 1: stress <= 3
+     * group 2: stress <= 5
+     * group 3: stress <= 7
+     *
+     * This method returns 3 and the bridge selector will consider bridges that
+     * are stressed <= 3.
+     *
+     * @param bridges
+     * @return
+     */
+    private double findStressLimit(Collection<Bridge> bridges)
+    {
+        List<Double> bridgesStress = bridges
+            .stream()
+            .map(Bridge::getStress)
+            .sorted()
+            .collect(Collectors.toList());
+
+        // We consider three groups, namely: "low-stressed", "medium-stressed"
+        // and "high-stressed" bridges.
+        int numberOfGroups = 3;
+
+        double min = bridgesStress.get(0);
+        double max = bridgesStress.get(bridgesStress.size() - 1);
+        double step = Math.ceil((max - min) / numberOfGroups);
+        return min + step;
+    }
+
     /**
      * Selects a bridge to be used for a new participant in a conference,
      * assuming that no other bridge is used by the conference (i.e. this
      * is the initial selection of a bridge for the conference).
      *
-     * @param bridges the list of bridges to select from.
+     * @param bridges the list of bridges to select from (ordered by the total
+     * bitrate that they're handling, which is used an indirect load indicator).
+     * @param conferenceBridges the bridges that are in the conference.
      * @param participantRegion the region of the participant for which a
      * bridge is to be selected.
      * @return the selected bridge, or {@code null} if no bridge is
      * available.
      */
-    private Bridge selectInitial(List<Bridge> bridges,
+    public Bridge doSelect(List<Bridge> bridges,
+                                 List<Bridge> conferenceBridges,
                                  String participantRegion)
     {
-        Bridge bridge = null;
-
-        // Prefer a bridge in the participant's region.
-        if (participantRegion != null)
+        if (bridges.isEmpty())
         {
-            bridge = findFirstOperationalInRegion(bridges, participantRegion);
+            return null;
         }
 
-        // Otherwise, prefer a bridge in the local region.
-        if (bridge == null)
-        {
-            bridge = findFirstOperationalInRegion(bridges, localRegion);
-        }
+        double maxStress = findStressLimit(bridges);
 
-        // Otherwise, just find the first operational bridge.
-        if (bridge == null)
-        {
-            bridge = findFirstOperationalInRegion(bridges, null);
-        }
+        // Try the first operational bridge in the participant region and
```

Comment:
"operational" has another meaning in jicofo. Maybe change it to "lightly loaded" or something that mentions stress?